rskj-core/src/main/java/co/rsk/peg/RepositoryBtcBlockStoreWithCache.java

```diff
@@ -132,7 +197,48 @@ public StoredBlock getStoredBlockAtMainChainHeight(int height) throws BlockStore
         StoredBlock chainHead = getChainHead();
         int depth = chainHead.getHeight() - height;
         logger.trace("Getting btc block at depth: {}", depth);
-        return getStoredBlockAtMainChainDepth(depth);
+
+        if (depth < 0) {
+            String message = String.format(
+                "Height provided is higher than chain head. provided: %n. chain head: %n",
+                height,
+                chainHead.getHeight()
+            );
+            logger.trace("[getStoredBlockAtMainChainHeight] {}", message);
+            throw new BlockStoreException(message);
+        }
+
+        if (activations.isActive(ConsensusRule.RSKIP199)) {
+            int btcHeightWhenBlockIndexActivates = this.bridgeConstants.getBtcHeightWhenBlockIndexActivates();
+            int maxDepthToSearch = this.bridgeConstants.getMaxDepthToSearchBlocksBelowIndexActivation();
+            int limit;
+            if (chainHead.getHeight() - btcHeightWhenBlockIndexActivates > maxDepthToSearch) {
+                limit = btcHeightWhenBlockIndexActivates;
+            } else {
+                limit = chainHead.getHeight() - maxDepthToSearch;
+            }
+            logger.trace("[getStoredBlockAtMainChainHeight] Chain head height is {} and the depth limit {}", chainHead.getHeight(), limit);
+
+            if (height < limit) {
+                String message = String.format(
```

Comment:
If `height < limit` this method is returning early instead of calling `getStoredBlockAtMainChainDepth`. Is this desired? I have no access to the rskip199 for checking this

Reply:
In fact there's this bug here where there will be an unwanted limit imposed by `getBtcHeightWhenBlockIndexActivates`. This is likely not what we want